gpiozero/output_devices.py

```diff
@@ -1289,12 +1291,14 @@ def backward(self, speed=1):
         """
         if not 0 <= speed <= 1:
             raise ValueError('backward speed must be between 0 and 1')
-        if isinstance(self.backward_device, DigitalOutputDevice) and isinstance(self.enable_device, DigitalOutputDevice):
+        if isinstance(self.backward_device, DigitalOutputDevice) and (
+            not hasattr(self, "enable_device") or isinstance(self.enable_device, DigitalOutputDevice)
+        ):
             if speed not in (0, 1):
                 raise ValueError(
                     'backward speed must be 0 or 1 with non-PWM Motors')
         self.forward_device.off()
-        if isinstance(self.enable_device, PWMOutputDevice):
+        if hasattr(self, "enable_device") and isinstance(self.enable_device, PWMOutputDevice):
```

Comment:
I wonder if it'd be worth having a `self.enable_is_pwm` boolean attribute and checking that, rather than all of these `hasattr(...) and isinstance(...)` tests? :man_shrugging:

Reply:
Sounds good. But I wonder if `DIGITAL`, `PWM`, `DIGITAL_WITH_DIGITAL_ENABLE`, `PWM_WITH_DIGITAL_ENABLE`, `DIGITAL_WITH_PWM_ENABLE` is a slightly clearer set of names?